Windows XP and PDH

I've been trying to get OSHI working with WinXP, but it looks like the last working release was 3.5.0. Anything newer than that gives me an error "Error looking up function 'PdhAddEnglishCounter': The specified procedure could not be found." I don't think all of the PDH APIs are available on WinXP. 

I've attached the results of mvn test if that helps at all.

```[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Operating System and Hardware Information
[INFO] oshi-core
[INFO] oshi-core-shaded
[INFO] oshi-json
[INFO] oshi-dist
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Operating System and Hardware Information 4.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- git-commit-id-plugin:2.2.5:revision (git-commit-id) @ oshi-parent ---
[INFO] 
[INFO] --- build-helper-maven-plugin:3.0.0:parse-version (default) @ oshi-parent ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven) @ oshi-parent ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:prepare-agent (pre-unit-test) @ oshi-parent ---
[INFO] surefireArgLine set to "-javaagent:C:\\Documents and Settings\\me\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.2\\org.jacoco.agent-0.8.2-runtime.jar=destfile=C:\\Documents and Settings\\me\\git\\oshi\\target\\jacoco.exec,excludes=**/impl/**:**/common/**:**/Abstract*/**:**/platform/**:**/linux/**:**/mac/**:**/windows/**:**/unix/**"
[INFO] 
[INFO] --- license-maven-plugin:3.0:format (default) @ oshi-parent ---
[INFO] Updating license headers...
[WARNING] Unknown file extension: C:/Documents and Settings/me/git/oshi/LICENSE_HEADER
[WARNING] Unable to find a comment style definition for some files. You may want to add a custom mapping for the relevant file extensions.
[INFO] 
[INFO] --- bnd-maven-plugin:4.1.0:bnd-process (default) @ oshi-parent ---
[INFO] skip project with packaging=pom
[INFO] 
[INFO] --- animal-sniffer-maven-plugin:1.17:check (test-sniffer) @ oshi-parent ---
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java17:1.0
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:report (post-unit-test) @ oshi-parent ---
[INFO] Skipping JaCoCo execution due to missing execution data file.
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M2:test (run-unit-tests) @ oshi-parent ---
[INFO] No tests to run.
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building oshi-core 4.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- git-commit-id-plugin:2.2.5:revision (git-commit-id) @ oshi-core ---
[INFO] 
[INFO] --- build-helper-maven-plugin:3.0.0:parse-version (default) @ oshi-core ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven) @ oshi-core ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:prepare-agent (pre-unit-test) @ oshi-core ---
[INFO] surefireArgLine set to "-javaagent:C:\\Documents and Settings\\me\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.2\\org.jacoco.agent-0.8.2-runtime.jar=destfile=C:\\Documents and Settings\\me\\git\\oshi\\oshi-core\\target\\jacoco.exec,excludes=**/impl/**:**/common/**:**/Abstract*/**:**/platform/**:**/linux/**:**/mac/**:**/windows/**:**/unix/**"
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ oshi-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Documents and Settings\me\git\oshi\oshi-core\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ oshi-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- license-maven-plugin:3.0:format (default) @ oshi-core ---
[INFO] Updating license headers...
[INFO] Updating license header in: C:/Documents and Settings/me/git/oshi/oshi-core/TempWmicBatchFile.bat
[INFO] 
[INFO] --- bnd-maven-plugin:4.1.0:bnd-process (default) @ oshi-core ---
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ oshi-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 4 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ oshi-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M2:test (default-test) @ oshi-core ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running CentralProcessorTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 8.89 s <<< FAILURE! - in CentralProcessorTest
[ERROR] testCentralProcessor  Time elapsed: 8.859 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at oshi.hardware.CentralProcessorTest.testCentralProcessor(CentralProcessorTest.java:50)
Caused by: com.sun.jna.platform.win32.PdhUtil$PdhException: Pdh call failed with error code 0xC0000BBD
	at oshi.hardware.CentralProcessorTest.testCentralProcessor(CentralProcessorTest.java:50)

[INFO] Running ComputerSystemTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.203 s - in ComputerSystemTest
[INFO] Running DisksTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.016 s <<< FAILURE! - in DisksTest
[ERROR] testDisks  Time elapsed: 0.016 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at oshi.hardware.DisksTest.testDisks(DisksTest.java:54)
Caused by: com.sun.jna.platform.win32.PdhUtil$PdhException: Pdh call failed with error code 0xC0000BBD
	at oshi.hardware.DisksTest.testDisks(DisksTest.java:54)

[INFO] Running DisplayTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.015 s - in DisplayTest
[INFO] Running GlobalMemoryTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0 s <<< FAILURE! - in GlobalMemoryTest
[ERROR] testGlobalMemory  Time elapsed: 0 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'PdhAddEnglishCounter': The specified procedure could not be found.

	at oshi.hardware.GlobalMemoryTest.testGlobalMemory(GlobalMemoryTest.java:44)

[INFO] Running NetworksTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.032 s - in NetworksTest
[INFO] Running PowerSourceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in PowerSourceTest
[INFO] Running SensorsTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0 s <<< FAILURE! - in SensorsTest
[ERROR] testSensors  Time elapsed: 0 s  <<< ERROR!
com.sun.jna.platform.win32.PdhUtil$PdhException: Pdh call failed with error code 0xC0000BBD
	at oshi.hardware.SensorsTest.testSensors(SensorsTest.java:42)

[INFO] Running SoundCardTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.11 s - in SoundCardTest
[INFO] Running UsbDeviceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.343 s - in UsbDeviceTest
[INFO] Running FileSystemTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.063 s <<< FAILURE! - in FileSystemTest
[ERROR] testFileSystem  Time elapsed: 0.063 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'PdhAddEnglishCounter': The specified procedure could not be found.

	at oshi.software.os.FileSystemTest.testFileSystem(FileSystemTest.java:51)

[INFO] Running NetworkParamsTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.156 s - in NetworkParamsTest
[INFO] Running OperatingSystemTest
[ERROR] Tests run: 5, Failures: 0, Errors: 5, Skipped: 0, Time elapsed: 0.203 s <<< FAILURE! - in OperatingSystemTest
[ERROR] testOperatingSystem  Time elapsed: 0.047 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testOperatingSystem(OperatingSystemTest.java:65)

[ERROR] testProcessQueryByList  Time elapsed: 0.031 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testProcessQueryByList(OperatingSystemTest.java:111)

[ERROR] testOSProcessSetters  Time elapsed: 0.047 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testOSProcessSetters(OperatingSystemTest.java:209)

[ERROR] testGetCommandLine  Time elapsed: 0.031 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testGetCommandLine(OperatingSystemTest.java:263)

[ERROR] testGetChildProcesses  Time elapsed: 0.047 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testGetChildProcesses(OperatingSystemTest.java:145)

[INFO] Running SystemInfoTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in SystemInfoTest
[INFO] Running EdidUtilTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in EdidUtilTest
[INFO] Running ExecutingCommandTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.047 s - in ExecutingCommandTest
[INFO] Running FileUtilTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in FileUtilTest
[INFO] Running FormatUtilTest
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in FormatUtilTest
[INFO] Running MapUtilTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in MapUtilTest
[INFO] Running ParseUtilTest
[INFO] Tests run: 27, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in ParseUtilTest
[INFO] Running StringUtilTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in StringUtilTest
[INFO] Running UtilTest
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.093 s <<< FAILURE! - in UtilTest
[ERROR] testSleep  Time elapsed: 0.078 s  <<< FAILURE!
java.lang.AssertionError
	at oshi.util.UtilTest.testSleep(UtilTest.java:41)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   UtilTest.testSleep:41
[ERROR] Errors: 
[ERROR]   CentralProcessorTest.testCentralProcessor:50 » ExceptionInInitializer
[ERROR]   DisksTest.testDisks:54 » ExceptionInInitializer
[ERROR]   GlobalMemoryTest.testGlobalMemory:44 » UnsatisfiedLink Error looking up functi...
[ERROR]   SensorsTest.testSensors:42 » Pdh Pdh call failed with error code 0xC0000BBD
[ERROR]   FileSystemTest.testFileSystem:51 » UnsatisfiedLink Error looking up function '...
[ERROR]   OperatingSystemTest.testGetChildProcesses:145 » UnsatisfiedLink Error looking ...
[ERROR]   OperatingSystemTest.testGetCommandLine:263 » UnsatisfiedLink Error looking up ...
[ERROR]   OperatingSystemTest.testOSProcessSetters:209 » UnsatisfiedLink Error looking u...
[ERROR]   OperatingSystemTest.testOperatingSystem:65 » UnsatisfiedLink Error looking up ...
[ERROR]   OperatingSystemTest.testProcessQueryByList:111 » UnsatisfiedLink Error looking...
[INFO] 
[ERROR] Tests run: 66, Failures: 1, Errors: 10, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Operating System and Hardware Information .......... SUCCESS [  2.282 s]
[INFO] oshi-core .......................................... FAILURE [ 15.453 s]
[INFO] oshi-core-shaded ................................... SKIPPED
[INFO] oshi-json .......................................... SKIPPED
[INFO] oshi-dist .......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 18.219 s
[INFO] Finished at: 2019-01-03T12:15:54+10:00
[INFO] Final Memory: 18M/49M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M2:test (default-test) on project oshi-core: There are test failures.
[ERROR] 
[ERROR] Please refer to C:\Documents and Settings\me\git\oshi\oshi-core\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :oshi-core```

I tried the latest snapshot and I'm still getting errors. The tests are also still failing for me:

```
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Operating System and Hardware Information
[INFO] oshi-core
[INFO] oshi-core-shaded
[INFO] oshi-json
[INFO] oshi-dist
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Operating System and Hardware Information 4.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- git-commit-id-plugin:2.2.5:revision (git-commit-id) @ oshi-parent ---
[INFO] 
[INFO] --- build-helper-maven-plugin:3.0.0:parse-version (default) @ oshi-parent ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven) @ oshi-parent ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:prepare-agent (pre-unit-test) @ oshi-parent ---
[INFO] surefireArgLine set to "-javaagent:C:\\Documents and Settings\\me\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.2\\org.jacoco.agent-0.8.2-runtime.jar=destfile=C:\\Documents and Settings\\me\\git\\oshi\\target\\jacoco.exec,excludes=**/impl/**:**/common/**:**/Abstract*/**:**/platform/**:**/linux/**:**/mac/**:**/windows/**:**/unix/**"
[INFO] 
[INFO] --- license-maven-plugin:3.0:format (default) @ oshi-parent ---
[INFO] Updating license headers...
[WARNING] Unknown file extension: C:/Documents and Settings/me/git/oshi/LICENSE_HEADER
[WARNING] Unable to find a comment style definition for some files. You may want to add a custom mapping for the relevant file extensions.
[INFO] 
[INFO] --- bnd-maven-plugin:4.1.0:bnd-process (default) @ oshi-parent ---
[INFO] skip project with packaging=pom
[INFO] 
[INFO] --- animal-sniffer-maven-plugin:1.17:check (test-sniffer) @ oshi-parent ---
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java17:1.0
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:report (post-unit-test) @ oshi-parent ---
[INFO] Skipping JaCoCo execution due to missing execution data file.
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M2:test (run-unit-tests) @ oshi-parent ---
[INFO] No tests to run.
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building oshi-core 4.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- git-commit-id-plugin:2.2.5:revision (git-commit-id) @ oshi-core ---
[INFO] 
[INFO] --- build-helper-maven-plugin:3.0.0:parse-version (default) @ oshi-core ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven) @ oshi-core ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:prepare-agent (pre-unit-test) @ oshi-core ---
[INFO] surefireArgLine set to "-javaagent:C:\\Documents and Settings\\me\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.2\\org.jacoco.agent-0.8.2-runtime.jar=destfile=C:\\Documents and Settings\\me\\git\\oshi\\oshi-core\\target\\jacoco.exec,excludes=**/impl/**:**/common/**:**/Abstract*/**:**/platform/**:**/linux/**:**/mac/**:**/windows/**:**/unix/**"
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ oshi-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Documents and Settings\me\git\oshi\oshi-core\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ oshi-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- license-maven-plugin:3.0:format (default) @ oshi-core ---
[INFO] Updating license headers...
[INFO] Updating license header in: C:/Documents and Settings/me/git/oshi/oshi-core/src/main/java/oshi/util/platform/windows/WmiQueryHandler.java
[INFO] 
[INFO] --- bnd-maven-plugin:4.1.0:bnd-process (default) @ oshi-core ---
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ oshi-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 4 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ oshi-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M2:test (default-test) @ oshi-core ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running CentralProcessorTest
[INFO] Running EdidUtilTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.062 s - in EdidUtilTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 8.672 s <<< FAILURE! - in CentralProcessorTest
[ERROR] testCentralProcessor  Time elapsed: 8.672 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at oshi.hardware.CentralProcessorTest.testCentralProcessor(CentralProcessorTest.java:50)
Caused by: com.sun.jna.platform.win32.PdhUtil$PdhException: Pdh call failed with error code 0xC0000BBD
	at oshi.hardware.CentralProcessorTest.testCentralProcessor(CentralProcessorTest.java:50)

[INFO] Running ComputerSystemTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.281 s - in ComputerSystemTest
[INFO] Running DisksTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0 s <<< FAILURE! - in DisksTest
[ERROR] testDisks  Time elapsed: 0 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at oshi.hardware.DisksTest.testDisks(DisksTest.java:54)
Caused by: com.sun.jna.platform.win32.PdhUtil$PdhException: Pdh call failed with error code 0xC0000BBD
	at oshi.hardware.DisksTest.testDisks(DisksTest.java:54)

[INFO] Running DisplayTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in DisplayTest
[INFO] Running GlobalMemoryTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.015 s <<< FAILURE! - in GlobalMemoryTest
[ERROR] testGlobalMemory  Time elapsed: 0.015 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'PdhAddEnglishCounter': The specified procedure could not be found.

	at oshi.hardware.GlobalMemoryTest.testGlobalMemory(GlobalMemoryTest.java:44)

[INFO] Running NetworksTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.016 s - in NetworksTest
[INFO] Running PowerSourceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in PowerSourceTest
[INFO] Running SensorsTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0 s <<< FAILURE! - in SensorsTest
[ERROR] testSensors  Time elapsed: 0 s  <<< ERROR!
com.sun.jna.platform.win32.PdhUtil$PdhException: Pdh call failed with error code 0xC0000BBD
	at oshi.hardware.SensorsTest.testSensors(SensorsTest.java:42)

[INFO] Running SoundCardTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.703 s - in SoundCardTest
[INFO] Running UsbDeviceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.219 s - in UsbDeviceTest
[INFO] Running FileSystemTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.063 s <<< FAILURE! - in FileSystemTest
[ERROR] testFileSystem  Time elapsed: 0.063 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'PdhAddEnglishCounter': The specified procedure could not be found.

	at oshi.software.os.FileSystemTest.testFileSystem(FileSystemTest.java:51)

[INFO] Running NetworkParamsTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.156 s - in NetworkParamsTest
[INFO] Running OperatingSystemTest
[ERROR] Tests run: 5, Failures: 0, Errors: 5, Skipped: 0, Time elapsed: 0.203 s <<< FAILURE! - in OperatingSystemTest
[ERROR] testOperatingSystem  Time elapsed: 0.031 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testOperatingSystem(OperatingSystemTest.java:65)

[ERROR] testProcessQueryByList  Time elapsed: 0.031 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testProcessQueryByList(OperatingSystemTest.java:111)

[ERROR] testOSProcessSetters  Time elapsed: 0.047 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testOSProcessSetters(OperatingSystemTest.java:209)

[ERROR] testGetCommandLine  Time elapsed: 0.047 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testGetCommandLine(OperatingSystemTest.java:263)

[ERROR] testGetChildProcesses  Time elapsed: 0.047 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'WTSEnumerateProcessesEx': The specified procedure could not be found.

	at oshi.software.os.OperatingSystemTest.testGetChildProcesses(OperatingSystemTest.java:145)

[INFO] Running SystemInfoTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in SystemInfoTest
[INFO] Running ExecutingCommandTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.062 s - in ExecutingCommandTest
[INFO] Running FileUtilTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.016 s - in FileUtilTest
[INFO] Running FormatUtilTest
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in FormatUtilTest
[INFO] Running MapUtilTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in MapUtilTest
[INFO] Running ParseUtilTest
[INFO] Tests run: 27, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in ParseUtilTest
[INFO] Running StringUtilTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in StringUtilTest
[INFO] Running UtilTest
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.344 s <<< FAILURE! - in UtilTest
[ERROR] testSleep  Time elapsed: 0.328 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class oshi.hardware.platform.windows.WindowsCentralProcessor
	at oshi.util.UtilTest.testSleep(UtilTest.java:53)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   CentralProcessorTest.testCentralProcessor:50 » ExceptionInInitializer
[ERROR]   DisksTest.testDisks:54 » ExceptionInInitializer
[ERROR]   GlobalMemoryTest.testGlobalMemory:44 » UnsatisfiedLink Error looking up functi...
[ERROR]   SensorsTest.testSensors:42 » Pdh Pdh call failed with error code 0xC0000BBD
[ERROR]   FileSystemTest.testFileSystem:51 » UnsatisfiedLink Error looking up function '...
[ERROR]   OperatingSystemTest.testGetChildProcesses:145 » UnsatisfiedLink Error looking ...
[ERROR]   OperatingSystemTest.testGetCommandLine:263 » UnsatisfiedLink Error looking up ...
[ERROR]   OperatingSystemTest.testOSProcessSetters:209 » UnsatisfiedLink Error looking u...
[ERROR]   OperatingSystemTest.testOperatingSystem:65 » UnsatisfiedLink Error looking up ...
[ERROR]   OperatingSystemTest.testProcessQueryByList:111 » UnsatisfiedLink Error looking...
[ERROR]   UtilTest.testSleep:53 » NoClassDefFound Could not initialize class oshi.hardwa...
[INFO] 
[ERROR] Tests run: 66, Failures: 0, Errors: 11, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Operating System and Hardware Information .......... SUCCESS [  2.453 s]
[INFO] oshi-core .......................................... FAILURE [ 16.485 s]
[INFO] oshi-core-shaded ................................... SKIPPED
[INFO] oshi-json .......................................... SKIPPED
[INFO] oshi-dist .......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 19.313 s
[INFO] Finished at: 2019-01-07T06:40:38+10:00
[INFO] Final Memory: 19M/53M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M2:test (default-test) on project oshi-core: There are test failures.
[ERROR] 
[ERROR] Please refer to C:\Documents and Settings\me\git\oshi\oshi-core\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :oshi-core
```

Never trust a bug reporter! :)

I was using the new jar, but something was obviously not working. I just pulled from your branch and ran the tests again and it's different now:

```
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Operating System and Hardware Information
[INFO] oshi-core
[INFO] oshi-core-shaded
[INFO] oshi-json
[INFO] oshi-dist
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building Operating System and Hardware Information 4.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- git-commit-id-plugin:2.2.5:revision (git-commit-id) @ oshi-parent ---
[INFO] 
[INFO] --- build-helper-maven-plugin:3.0.0:parse-version (default) @ oshi-parent ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven) @ oshi-parent ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:prepare-agent (pre-unit-test) @ oshi-parent ---
[INFO] surefireArgLine set to "-javaagent:C:\\Documents and Settings\\me\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.2\\org.jacoco.agent-0.8.2-runtime.jar=destfile=C:\\Documents and Settings\\me\\git\\oshi\\target\\jacoco.exec,excludes=**/impl/**:**/common/**:**/Abstract*/**:**/platform/**:**/linux/**:**/mac/**:**/windows/**:**/unix/**"
[INFO] 
[INFO] --- license-maven-plugin:3.0:format (default) @ oshi-parent ---
[INFO] Updating license headers...
[WARNING] Unknown file extension: C:/Documents and Settings/me/git/oshi/LICENSE_HEADER
[WARNING] Unable to find a comment style definition for some files. You may want to add a custom mapping for the relevant file extensions.
[INFO] 
[INFO] --- bnd-maven-plugin:4.1.0:bnd-process (default) @ oshi-parent ---
[INFO] skip project with packaging=pom
[INFO] 
[INFO] --- animal-sniffer-maven-plugin:1.17:check (test-sniffer) @ oshi-parent ---
[INFO] Checking unresolved references to org.codehaus.mojo.signature:java17:1.0
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:report (post-unit-test) @ oshi-parent ---
[INFO] Skipping JaCoCo execution due to missing execution data file.
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M2:test (run-unit-tests) @ oshi-parent ---
[INFO] No tests to run.
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building oshi-core 4.0.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- git-commit-id-plugin:2.2.5:revision (git-commit-id) @ oshi-core ---
[INFO] 
[INFO] --- build-helper-maven-plugin:3.0.0:parse-version (default) @ oshi-core ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-maven) @ oshi-core ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:prepare-agent (pre-unit-test) @ oshi-core ---
[INFO] surefireArgLine set to "-javaagent:C:\\Documents and Settings\\me\\.m2\\repository\\org\\jacoco\\org.jacoco.agent\\0.8.2\\org.jacoco.agent-0.8.2-runtime.jar=destfile=C:\\Documents and Settings\\me\\git\\oshi\\oshi-core\\target\\jacoco.exec,excludes=**/impl/**:**/common/**:**/Abstract*/**:**/platform/**:**/linux/**:**/mac/**:**/windows/**:**/unix/**"
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ oshi-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Documents and Settings\me\git\oshi\oshi-core\src\main\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ oshi-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- license-maven-plugin:3.0:format (default) @ oshi-core ---
[INFO] Updating license headers...
[INFO] Updating license header in: C:/Documents and Settings/me/git/oshi/oshi-core/TempWmicBatchFile.bat
[INFO] 
[INFO] --- bnd-maven-plugin:4.1.0:bnd-process (default) @ oshi-core ---
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ oshi-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 4 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ oshi-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M2:test (default-test) @ oshi-core ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running CentralProcessorTest
[INFO] Running NetworkParamsTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 8.843 s <<< FAILURE! - in CentralProcessorTest
[ERROR] testCentralProcessor  Time elapsed: 8.843 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'PdhCollectQueryDataWithTime': The specified procedure could not be found.

	at com.sun.jna.Function.<init>(Function.java:251)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:594)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:570)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:556)
	at com.sun.jna.Library$Handler.invoke(Library.java:228)
	at com.sun.proxy.$Proxy10.PdhCollectQueryDataWithTime(Unknown Source)
	at oshi.util.platform.windows.PerfDataUtil.updateQueryTimestamp(PerfDataUtil.java:380)
	at oshi.util.platform.windows.PerfDataUtil.updateQuery(PerfDataUtil.java:200)
	at oshi.hardware.platform.windows.WindowsCentralProcessor.refreshTickCounters(WindowsCentralProcessor.java:463)
	at oshi.hardware.platform.windows.WindowsCentralProcessor.getSystemCpuLoadTicks(WindowsCentralProcessor.java:320)
	at oshi.hardware.common.AbstractCentralProcessor.updateSystemTicks(AbstractCentralProcessor.java:407)
	at oshi.hardware.common.AbstractCentralProcessor.getSystemCpuLoadBetweenTicks(AbstractCentralProcessor.java:383)
	at oshi.hardware.CentralProcessorTest.testCentralProcessor(CentralProcessorTest.java:84)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:39)
	at org.junit.vintage.engine.VintageTestEngine$$Lambda$129/16840200.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:79)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:70)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$108/18885178.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[INFO] Running ComputerSystemTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.125 s - in ComputerSystemTest
[INFO] Running DisksTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.156 s - in DisksTest
[INFO] Running DisplayTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in DisplayTest
[INFO] Running GlobalMemoryTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0 s <<< FAILURE! - in GlobalMemoryTest
[ERROR] testGlobalMemory  Time elapsed: 0 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'PdhCollectQueryDataWithTime': The specified procedure could not be found.

	at com.sun.jna.Function.<init>(Function.java:251)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:594)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:570)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:556)
	at com.sun.jna.Library$Handler.invoke(Library.java:228)
	at com.sun.proxy.$Proxy10.PdhCollectQueryDataWithTime(Unknown Source)
	at oshi.util.platform.windows.PerfDataUtil.updateQueryTimestamp(PerfDataUtil.java:380)
	at oshi.util.platform.windows.PerfDataUtil.updateQuery(PerfDataUtil.java:179)
	at oshi.hardware.platform.windows.WindowsGlobalMemory.updateMeminfo(WindowsGlobalMemory.java:138)
	at oshi.hardware.common.AbstractGlobalMemory.getTotal(AbstractGlobalMemory.java:71)
	at oshi.hardware.GlobalMemoryTest.testGlobalMemory(GlobalMemoryTest.java:48)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:39)
	at org.junit.vintage.engine.VintageTestEngine$$Lambda$129/16840200.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:79)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:70)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$108/18885178.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[INFO] Running NetworksTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.031 s - in NetworksTest
[INFO] Running PowerSourceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in PowerSourceTest
[INFO] Running SensorsTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.078 s - in SensorsTest
[INFO] Running SoundCardTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.344 s - in SoundCardTest
[INFO] Running UsbDeviceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.203 s - in UsbDeviceTest
[INFO] Running FileSystemTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.078 s <<< FAILURE! - in FileSystemTest
[ERROR] testFileSystem  Time elapsed: 0.078 s  <<< ERROR!
java.lang.UnsatisfiedLinkError: 
Error looking up function 'PdhCollectQueryDataWithTime': The specified procedure could not be found.

	at com.sun.jna.Function.<init>(Function.java:251)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:594)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:570)
	at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:556)
	at com.sun.jna.Library$Handler.invoke(Library.java:228)
	at com.sun.proxy.$Proxy10.PdhCollectQueryDataWithTime(Unknown Source)
	at oshi.util.platform.windows.PerfDataUtil.updateQueryTimestamp(PerfDataUtil.java:380)
	at oshi.util.platform.windows.PerfDataUtil.updateQuery(PerfDataUtil.java:179)
	at oshi.software.os.windows.WindowsFileSystem.getOpenFileDescriptors(WindowsFileSystem.java:288)
	at oshi.software.os.FileSystemTest.testFileSystem(FileSystemTest.java:52)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:39)
	at org.junit.vintage.engine.VintageTestEngine$$Lambda$129/16840200.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:79)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:70)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$108/18885178.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[INFO] Running OperatingSystemTest
[ERROR] Tests run: 5, Failures: 0, Errors: 5, Skipped: 0, Time elapsed: 1.375 s <<< FAILURE! - in OperatingSystemTest
[ERROR] testOperatingSystem  Time elapsed: 0.282 s  <<< ERROR!
java.lang.NullPointerException
	at oshi.software.os.windows.WindowsOperatingSystem.processMapToList(WindowsOperatingSystem.java:389)
	at oshi.software.os.windows.WindowsOperatingSystem.getProcesses(WindowsOperatingSystem.java:281)
	at oshi.software.common.AbstractOperatingSystem.getProcesses(AbstractOperatingSystem.java:188)
	at oshi.software.os.OperatingSystemTest.testOperatingSystem(OperatingSystemTest.java:65)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:39)
	at org.junit.vintage.engine.VintageTestEngine$$Lambda$129/16840200.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:79)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:70)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$108/18885178.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[ERROR] testProcessQueryByList  Time elapsed: 0.359 s  <<< ERROR!
java.lang.NullPointerException
	at oshi.software.os.windows.WindowsOperatingSystem.processMapToList(WindowsOperatingSystem.java:389)
	at oshi.software.os.windows.WindowsOperatingSystem.getProcesses(WindowsOperatingSystem.java:281)
	at oshi.software.common.AbstractOperatingSystem.getProcesses(AbstractOperatingSystem.java:188)
	at oshi.software.os.OperatingSystemTest.testProcessQueryByList(OperatingSystemTest.java:111)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:39)
	at org.junit.vintage.engine.VintageTestEngine$$Lambda$129/16840200.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:79)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:70)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$108/18885178.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[ERROR] testOSProcessSetters  Time elapsed: 0.078 s  <<< ERROR!
java.lang.NullPointerException
	at oshi.software.os.windows.WindowsOperatingSystem.processMapToList(WindowsOperatingSystem.java:389)
	at oshi.software.os.windows.WindowsOperatingSystem.getProcess(WindowsOperatingSystem.java:320)
	at oshi.software.os.windows.WindowsOperatingSystem.getProcess(WindowsOperatingSystem.java:314)
	at oshi.software.os.OperatingSystemTest.testOSProcessSetters(OperatingSystemTest.java:209)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:39)
	at org.junit.vintage.engine.VintageTestEngine$$Lambda$129/16840200.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:79)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:70)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$108/18885178.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[ERROR] testGetCommandLine  Time elapsed: 0.266 s  <<< ERROR!
java.lang.NullPointerException
	at oshi.software.os.windows.WindowsOperatingSystem.processMapToList(WindowsOperatingSystem.java:389)
	at oshi.software.os.windows.WindowsOperatingSystem.getProcesses(WindowsOperatingSystem.java:281)
	at oshi.software.common.AbstractOperatingSystem.getProcesses(AbstractOperatingSystem.java:188)
	at oshi.software.os.OperatingSystemTest.testGetCommandLine(OperatingSystemTest.java:263)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:39)
	at org.junit.vintage.engine.VintageTestEngine$$Lambda$129/16840200.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:79)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:70)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$108/18885178.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[ERROR] testGetChildProcesses  Time elapsed: 0.39 s  <<< ERROR!
java.lang.NullPointerException
	at oshi.software.os.windows.WindowsOperatingSystem.processMapToList(WindowsOperatingSystem.java:389)
	at oshi.software.os.windows.WindowsOperatingSystem.getProcesses(WindowsOperatingSystem.java:281)
	at oshi.software.common.AbstractOperatingSystem.getProcesses(AbstractOperatingSystem.java:188)
	at oshi.software.os.OperatingSystemTest.testGetChildProcesses(OperatingSystemTest.java:145)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:39)
	at org.junit.vintage.engine.VintageTestEngine$$Lambda$129/16840200.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:79)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:70)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$108/18885178.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[INFO] Running SystemInfoTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in SystemInfoTest
[INFO] Running EdidUtilTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in EdidUtilTest
[INFO] Running FileUtilTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in FileUtilTest
[INFO] Running FormatUtilTest
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in FormatUtilTest
[INFO] Running MapUtilTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in MapUtilTest
[INFO] Running UtilTest
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.296 s <<< FAILURE! - in UtilTest
[ERROR] testSleep  Time elapsed: 0.281 s  <<< FAILURE!
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at oshi.util.UtilTest.testSleep(UtilTest.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:39)
	at org.junit.vintage.engine.VintageTestEngine$$Lambda$129/16840200.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(Unknown Source)
	at java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.util.Iterator.forEachRemaining(Unknown Source)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(Unknown Source)
	at java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.util.stream.ReferencePipeline.forEach(Unknown Source)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:79)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:70)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
	at org.junit.platform.launcher.core.DefaultLauncher$$Lambda$108/18885178.accept(Unknown Source)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:109)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)

[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 58.813 s - in NetworkParamsTest
[INFO] Running ExecutingCommandTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.063 s - in ExecutingCommandTest
[INFO] Running ParseUtilTest
[INFO] Tests run: 27, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.015 s - in ParseUtilTest
[INFO] Running StringUtilTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in StringUtilTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   UtilTest.testSleep:46
[ERROR] Errors: 
[ERROR]   CentralProcessorTest.testCentralProcessor:84 » UnsatisfiedLink Error looking u...
[ERROR]   GlobalMemoryTest.testGlobalMemory:48 » UnsatisfiedLink Error looking up functi...
[ERROR]   FileSystemTest.testFileSystem:52 » UnsatisfiedLink Error looking up function '...
[ERROR]   OperatingSystemTest.testGetChildProcesses:145 » NullPointer
[ERROR]   OperatingSystemTest.testGetCommandLine:263 » NullPointer
[ERROR]   OperatingSystemTest.testOSProcessSetters:209 » NullPointer
[ERROR]   OperatingSystemTest.testOperatingSystem:65 » NullPointer
[ERROR]   OperatingSystemTest.testProcessQueryByList:111 » NullPointer
[INFO] 
[ERROR] Tests run: 66, Failures: 1, Errors: 8, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Operating System and Hardware Information .......... SUCCESS [  2.234 s]
[INFO] oshi-core .......................................... FAILURE [01:02 min]
[INFO] oshi-core-shaded ................................... SKIPPED
[INFO] oshi-json .......................................... SKIPPED
[INFO] oshi-dist .......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:04 min
[INFO] Finished at: 2019-01-07T09:37:34+10:00
[INFO] Final Memory: 19M/48M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M2:test (default-test) on project oshi-core: There are test failures.
[ERROR] 
[ERROR] Please refer to C:\Documents and Settings\me\git\oshi\oshi-core\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :oshi-core
```

--- a/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsCentralProcessor.java
+++ b/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsCentralProcessor.java
@@ -45,6 +45,7 @@ import com.sun.jna.platform.win32.COM.WbemcliUtil.WmiQuery;
 import com.sun.jna.platform.win32.COM.WbemcliUtil.WmiResult;
 
 import oshi.hardware.common.AbstractCentralProcessor;
+import oshi.util.platform.windows.PdhUtilXP;
 import oshi.util.platform.windows.PerfDataUtil;
 import oshi.util.platform.windows.PerfDataUtil.PerfCounter;
 import oshi.util.platform.windows.WmiQueryHandler;
@@ -70,23 +71,8 @@ public class WindowsCentralProcessor extends AbstractCentralProcessor {
     // Save Windows version info for 32 bit/64 bit branch later
     private static final byte MAJOR_VERSION = Kernel32.INSTANCE.GetVersion().getLow().byteValue();
 
-    // Localize the "Processor" counter string. English counter names should
-    // normally be in HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows
-    // NT\CurrentVersion\Perflib\009\Counter, but language manipulations may
-    // delete the 009 index. In this case we can assume English must be the
-    // language and continue. We may still fail to match the name if the
-    // assumption is wrong but it's better than nothing.
     private static final String PROCESSOR = "Processor";
-    private static final String PROCESSOR_LOCALIZED;
-    static {
-        String localized = null;
-        try {
-            localized = PdhUtil.PdhLookupPerfNameByIndex(null, PdhUtil.PdhLookupPerfIndexByEnglishName(PROCESSOR));
-        } catch (Win32Exception e) {
-            LOG.error("Unable to locate English counter names in registry Perflib 009. Assuming English counters.");
-        }
-        PROCESSOR_LOCALIZED = (localized != null && localized.length() > 0) ? localized : PROCESSOR;
-    }
+    private static String processorLocalized = null;
     private static final String TOTAL_INSTANCE = "_Total";
 
     /*
@@ -192,7 +178,7 @@ public class WindowsCentralProcessor extends AbstractCentralProcessor {
 
         boolean enumeration = true;
         try {
-            PdhEnumObjectItems objectItems = PdhUtil.PdhEnumObjectItems(null, null, PROCESSOR_LOCALIZED, 100);
+            PdhEnumObjectItems objectItems = PdhUtil.PdhEnumObjectItems(null, null, getProcessorLocalized(), 100);
 
             if (!objectItems.getInstances().isEmpty()) {
                 // % Processor Time is actually Idle time
@@ -257,7 +243,7 @@ public class WindowsCentralProcessor extends AbstractCentralProcessor {
                 }
             }
         } catch (PdhException e) {
-            LOG.warn("Unable to enumerate performance counter instances for {}.", PROCESSOR_LOCALIZED);
+            LOG.warn("Unable to enumerate performance counter instances for {}.", getProcessorLocalized());
             enumeration = false;
         }
         if (!enumeration) {
@@ -477,4 +463,28 @@ public class WindowsCentralProcessor extends AbstractCentralProcessor {
             lastRefresh = PerfDataUtil.updateQuery(PROCESSOR);
         }
     }
+
+    /**
+     * Localize the "Processor" counter string. English counter names should
+     * normally be in HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows
+     * NT\CurrentVersion\Perflib\009\Counter, but language manipulations may
+     * delete the 009 index. In this case we can assume English must be the
+     * language and continue. We may still fail to match the name if the
+     * assumption is wrong but it's better than nothing.
+     */
+    private static String getProcessorLocalized() {
+        if (processorLocalized == null) {
+            try {
+                processorLocalized = PdhUtilXP.PdhLookupPerfNameByIndex(null,
+                        PdhUtil.PdhLookupPerfIndexByEnglishName(PROCESSOR));
+            } catch (Win32Exception e) {
+                LOG.error("Unable to locate English counter names in registry Perflib 009. Assuming English counters.");
+            }
+            if (processorLocalized == null || processorLocalized.length() == 0) {
+                processorLocalized = PROCESSOR;
+            }
+        }
+        return processorLocalized;
+    }
+
 }

--- a/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsDisks.java
+++ b/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsDisks.java
@@ -47,6 +47,7 @@ import oshi.hardware.HWDiskStore;
 import oshi.hardware.HWPartition;
 import oshi.util.MapUtil;
 import oshi.util.ParseUtil;
+import oshi.util.platform.windows.PdhUtilXP;
 import oshi.util.platform.windows.PerfDataUtil;
 import oshi.util.platform.windows.PerfDataUtil.PerfCounter;
 import oshi.util.platform.windows.WmiQueryHandler;
@@ -79,7 +80,7 @@ public class WindowsDisks implements Disks {
 
     private static final String PHYSICALDRIVE_PREFIX = "\\\\.\\PHYSICALDRIVE";
     private static final String PHYSICAL_DISK = "PhysicalDisk";
-    private static final String PHYSICAL_DISK_LOCALIZED = PdhUtil.PdhLookupPerfNameByIndex(null,
+    private static final String PHYSICAL_DISK_LOCALIZED = PdhUtilXP.PdhLookupPerfNameByIndex(null,
             PdhUtil.PdhLookupPerfIndexByEnglishName(PHYSICAL_DISK));
     private static final String TOTAL_INSTANCE = "_Total";
 
@@ -128,7 +129,7 @@ public class WindowsDisks implements Disks {
     private static WmiQuery<PhysicalDiskProperty> physicalDiskQuery = null;
 
     static {
-        String physicalDisk = PdhUtil.PdhLookupPerfNameByIndex(null,
+        String physicalDisk = PdhUtilXP.PdhLookupPerfNameByIndex(null,
                 PdhUtil.PdhLookupPerfIndexByEnglishName(PHYSICAL_DISK));
         boolean enumeration = true;
         try {

--- a/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsSensors.java
+++ b/oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsSensors.java
@@ -33,6 +33,7 @@ import com.sun.jna.platform.win32.COM.WbemcliUtil.WmiQuery;
 import com.sun.jna.platform.win32.COM.WbemcliUtil.WmiResult;
 
 import oshi.hardware.Sensors;
+import oshi.util.platform.windows.PdhUtilXP;
 import oshi.util.platform.windows.PerfDataUtil;
 import oshi.util.platform.windows.PerfDataUtil.PerfCounter;
 import oshi.util.platform.windows.WmiQueryHandler;
@@ -90,7 +91,7 @@ public class WindowsSensors implements Sensors {
     }
 
     private void initPdhCounters() {
-        String thermalZoneInfo = PdhUtil.PdhLookupPerfNameByIndex(null,
+        String thermalZoneInfo = PdhUtilXP.PdhLookupPerfNameByIndex(null,
                 PdhUtil.PdhLookupPerfIndexByEnglishName("Thermal Zone Information"));
         boolean enumeration = false;
         if (!thermalZoneInfo.isEmpty()) {

--- a/oshi-core/src/main/java/oshi/jna/platform/windows/Kernel32.java
+++ b/oshi-core/src/main/java/oshi/jna/platform/windows/Kernel32.java
@@ -1,0 +1,117 @@
+/**
+ * OSHI (https://github.com/oshi/oshi)
+ *
+ * Copyright (c) 2010 - 2019 The OSHI Project Team:
+ * https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.jna.platform.windows;
+
+import com.sun.jna.Native;
+
+/**
+ * Kernel32 and WinNT info submitted to JNA as #1050. This class should be
+ * considered non-API as it may be removed if/when its code is incorporated into
+ * the JNA project.
+ */
+public interface Kernel32 extends com.sun.jna.platform.win32.Kernel32 {
+    Kernel32 INSTANCE = Native.load("Kernel32", Kernel32.class);
+
+    /*
+     * WIN32_WINNT version constants
+     */
+    short WIN32_WINNT_NT4 = 0x0400; // Windows NT 4.0
+    short WIN32_WINNT_WIN2K = 0x0500; // Windows 2000
+    short WIN32_WINNT_WINXP = 0x0501; // Windows XP
+    short WIN32_WINNT_WS03 = 0x0502; // Windows Server 2003
+    short WIN32_WINNT_WIN6 = 0x0600; // Windows Vista
+    short WIN32_WINNT_VISTA = 0x0600; // Windows Vista
+    short WIN32_WINNT_WS08 = 0x0600; // Windows Server 2008
+    short WIN32_WINNT_LONGHORN = 0x0600; // Windows Vista
+    short WIN32_WINNT_WIN7 = 0x0601; // Windows 7
+    short WIN32_WINNT_WIN8 = 0x0602; // Windows 8
+    short WIN32_WINNT_WINBLUE = 0x0603; // Windows 8.1
+    short WIN32_WINNT_WINTHRESHOLD = 0x0A00; // Windows 10
+    short WIN32_WINNT_WIN10 = 0x0A00; // Windows 10
+
+    /**
+     * Compares a set of operating system version requirements to the
+     * corresponding values for the currently running version of the system.This
+     * function is subject to manifest-based behavior.
+     * 
+     * @param lpVersionInformation
+     *            A pointer to an OSVERSIONINFOEX structure containing the
+     *            operating system version requirements to compare. The
+     *            dwTypeMask parameter indicates the members of this structure
+     *            that contain information to compare.
+     * 
+     *            You must set the dwOSVersionInfoSize member of this structure
+     *            to sizeof(OSVERSIONINFOEX). You must also specify valid data
+     *            for the members indicated by dwTypeMask. The function ignores
+     *            structure members for which the corresponding dwTypeMask bit
+     *            is not set.
+     * @param dwTypeMask
+     *            A mask that indicates the members of the OSVERSIONINFOEX
+     *            structure to be tested.
+     * @param dwlConditionMask
+     *            The type of comparison to be used for each lpVersionInfo
+     *            member being compared. To build this value, call the
+     *            VerSetConditionMask function once for each OSVERSIONINFOEX
+     *            member being compared.
+     * @return If the currently running operating system satisfies the specified
+     *         requirements, the return value is a nonzero value.
+     * 
+     *         If the current system does not satisfy the requirements, the
+     *         return value is zero and GetLastError returns
+     *         ERROR_OLD_WIN_VERSION.
+     * 
+     *         If the function fails, the return value is zero and GetLastError
+     *         returns an error code other than ERROR_OLD_WIN_VERSION.
+     */
+    boolean VerifyVersionInfoW(OSVERSIONINFOEX lpVersionInformation, int dwTypeMask, long dwlConditionMask);
+
+    /**
+     * Sets the bits of a 64-bit value to indicate the comparison operator to
+     * use for a specified operating system version attribute. This function is
+     * used to build the dwlConditionMask parameter of the VerifyVersionInfo
+     * function.
+     * 
+     * @param conditionMask
+     *            A value to be passed as the dwlConditionMask parameter of the
+     *            VerifyVersionInfo function. The function stores the comparison
+     *            information in the bits of this variable.
+     * 
+     *            Before the first call to VerSetCondition, initialize this
+     *            variable to zero. For subsequent calls, pass in the variable
+     *            used in the previous call.
+     * @param typeMask
+     *            A mask that indicates the member of the OSVERSIONINFOEX
+     *            structure whose comparison operator is being set. This value
+     *            corresponds to one of the bits specified in the dwTypeMask
+     *            parameter for the VerifyVersionInfo function.
+     * @param condition
+     *            The operator to be used for the comparison. The
+     *            VerifyVersionInfo function uses this operator to compare a
+     *            specified attribute value to the corresponding value for the
+     *            currently running system.
+     * @return The function returns the condition mask value.
+     */
+    long VerSetConditionMask(long conditionMask, int typeMask, byte condition);
+
+}

--- a/oshi-core/src/main/java/oshi/jna/platform/windows/VersionHelpers.java
+++ b/oshi-core/src/main/java/oshi/jna/platform/windows/VersionHelpers.java
@@ -1,0 +1,214 @@
+/**
+ * OSHI (https://github.com/oshi/oshi)
+ *
+ * Copyright (c) 2010 - 2019 The OSHI Project Team:
+ * https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.jna.platform.windows;
+
+import com.sun.jna.platform.win32.WinDef.DWORD;
+import com.sun.jna.platform.win32.WinDef.WORD;
+import com.sun.jna.platform.win32.WinNT;
+import com.sun.jna.platform.win32.WinNT.OSVERSIONINFOEX;
+
+/**
+ * The following functions can be used to determine the current operating system
+ * version or identify whether it is a Windows or Windows Server release. These
+ * functions provide simple tests that use the VerifyVersionInfo function and
+ * the recommended greater than or equal to comparisons that are proven as a
+ * robust means to determine the operating system version.
+ */
+public class VersionHelpers {
+
+    /**
+     * This function is useful in confirming a version of Windows Server that
+     * doesn't share a version number with a client release. You should only use
+     * this function if the other provided version helper functions do not fit
+     * your scenario.
+     * 
+     * @param wMajorVersion
+     *            The major version to test
+     * @param wMinorVersion
+     *            The minor version to test
+     * @param wServicePackMajor
+     *            The service pack to test
+     * @return True if the current OS version matches, or is greater than, the
+     *         provided version information.
+     */
+    public static boolean IsWindowsVersionOrGreater(int wMajorVersion, int wMinorVersion, int wServicePackMajor) {
+        OSVERSIONINFOEX osvi = new OSVERSIONINFOEX();
+        osvi.dwOSVersionInfoSize = new DWORD(osvi.size());
+        osvi.dwMajorVersion = new DWORD(wMajorVersion);
+        osvi.dwMinorVersion = new DWORD(wMinorVersion);
+        osvi.wServicePackMajor = new WORD(wServicePackMajor);
+
+        long dwlConditionMask = 0;
+        dwlConditionMask = Kernel32.INSTANCE.VerSetConditionMask(dwlConditionMask, WinNT.VER_MAJORVERSION,
+                (byte) WinNT.VER_GREATER_EQUAL);
+        dwlConditionMask = Kernel32.INSTANCE.VerSetConditionMask(dwlConditionMask, WinNT.VER_MINORVERSION,
+                (byte) WinNT.VER_GREATER_EQUAL);
+        dwlConditionMask = Kernel32.INSTANCE.VerSetConditionMask(dwlConditionMask, WinNT.VER_SERVICEPACKMAJOR,
+                (byte) WinNT.VER_GREATER_EQUAL);
+
+        return Kernel32.INSTANCE.VerifyVersionInfoW(osvi,
+                WinNT.VER_MAJORVERSION | WinNT.VER_MINORVERSION | WinNT.VER_SERVICEPACKMAJOR, dwlConditionMask);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows XP version.
+     */
+    public static boolean IsWindowsXPOrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_WINXP), LOBYTE(Kernel32.WIN32_WINNT_WINXP), 0);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows XP with Service Pack 1 (SP1) version.
+     */
+    public static boolean IsWindowsXPSP1OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_WINXP), LOBYTE(Kernel32.WIN32_WINNT_WINXP), 1);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows XP with Service Pack 2 (SP2) version.
+     */
+    public static boolean IsWindowsXPSP2OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_WINXP), LOBYTE(Kernel32.WIN32_WINNT_WINXP), 2);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows XP with Service Pack 3 (SP3) version.
+     */
+    public static boolean IsWindowsXPSP3OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_WINXP), LOBYTE(Kernel32.WIN32_WINNT_WINXP), 3);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows Vista version.
+     */
+    public static boolean IsWindowsVistaOrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_VISTA), LOBYTE(Kernel32.WIN32_WINNT_VISTA), 0);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows Vista with Service Pack 1 (SP1) version.
+     */
+    public static boolean IsWindowsVistaSP1OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_VISTA), LOBYTE(Kernel32.WIN32_WINNT_VISTA), 1);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows Vista with Service Pack 2 (SP2) version.
+     */
+    public static boolean IsWindowsVistaSP2OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_VISTA), LOBYTE(Kernel32.WIN32_WINNT_VISTA), 2);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows 7 version.
+     */
+    public static boolean IsWindows7OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_WIN7), LOBYTE(Kernel32.WIN32_WINNT_WIN7), 0);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows 7 with Service Pack 1 (SP1) version.
+     */
+    public static boolean IsWindows7SP1OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_WIN7), LOBYTE(Kernel32.WIN32_WINNT_WIN7), 1);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows 8 version.
+     */
+    public static boolean IsWindows8OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_WIN8), LOBYTE(Kernel32.WIN32_WINNT_WIN8), 0);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows 8.1 version. For Windows 10, IsWindows8Point1OrGreater
+     *         returns false unless the application contains a manifest that
+     *         includes a compatibility section that contains the GUIDs that
+     *         designate Windows 8.1 and/or Windows 10.
+     */
+    public static boolean IsWindows8Point1OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_WINBLUE), LOBYTE(Kernel32.WIN32_WINNT_WINBLUE), 0);
+    }
+
+    /**
+     * @return true if the current OS version matches, or is greater than, the
+     *         Windows 10 version. For Windows 10, IsWindows10OrGreater returns
+     *         false unless the application contains a manifest that includes a
+     *         compatibility section that contains the GUID that designates
+     *         Windows 10.
+     */
+    public static boolean IsWindows10OrGreater() {
+        return IsWindowsVersionOrGreater(HIBYTE(Kernel32.WIN32_WINNT_WIN10), LOBYTE(Kernel32.WIN32_WINNT_WIN10), 0);
+    }
+
+    /**
+     * Applications that need to distinguish between server and client versions
+     * of Windows should call this function.
+     * 
+     * @return true if the current OS is a Windows Server release.
+     */
+    public static boolean IsWindowsServer() {
+        OSVERSIONINFOEX osvi = new OSVERSIONINFOEX();
+        osvi.dwOSVersionInfoSize = new DWORD(osvi.size());
+        osvi.wProductType = WinNT.VER_NT_WORKSTATION;
+
+        long dwlConditionMask = Kernel32.INSTANCE.VerSetConditionMask(0, WinNT.VER_PRODUCT_TYPE,
+                (byte) WinNT.VER_EQUAL);
+
+        return !Kernel32.INSTANCE.VerifyVersionInfoW(osvi, WinNT.VER_PRODUCT_TYPE, dwlConditionMask);
+    }
+
+    /**
+     * Get the high byte
+     * 
+     * @param word
+     *            a two-byte value
+     * @return The most significant byte
+     */
+    private static byte HIBYTE(short word) {
+        return (byte) ((word >> 8) & 0xFF);
+    }
+
+    /**
+     * Get the low byte
+     * 
+     * @param word
+     *            a two-byte value
+     * @return The least significant byte
+     */
+    private static byte LOBYTE(short word) {
+        return (byte) word;
+    }
+}

--- a/oshi-core/src/main/java/oshi/software/os/windows/WindowsOperatingSystem.java
+++ b/oshi-core/src/main/java/oshi/software/os/windows/WindowsOperatingSystem.java
@@ -55,6 +55,7 @@ import com.sun.jna.platform.win32.WinError;
 import com.sun.jna.platform.win32.WinNT;
 import com.sun.jna.platform.win32.WinNT.HANDLE;
 import com.sun.jna.platform.win32.WinNT.HANDLEByReference;
+import com.sun.jna.platform.win32.WinNT.LARGE_INTEGER;
 import com.sun.jna.platform.win32.WinPerf.PERF_COUNTER_BLOCK;
 import com.sun.jna.platform.win32.WinPerf.PERF_COUNTER_DEFINITION;
 import com.sun.jna.platform.win32.WinPerf.PERF_DATA_BLOCK;
@@ -91,6 +92,13 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
 
     private static final String PROCESS_BASE_CLASS = "Win32_Process";
     private static final WmiQuery<ProcessProperty> PROCESS_QUERY = new WmiQuery<>(null, ProcessProperty.class);
+
+    // Properties to get from WMI if WTSEnumerateProcesses doesn't work
+    enum ProcessXPProperty {
+        PROCESSID, NAME, KERNELMODETIME, USERMODETIME, THREADCOUNT, PAGEFILEUSAGE, HANDLECOUNT;
+    }
+
+    private static final WmiQuery<ProcessXPProperty> PROCESS_QUERY_XP = new WmiQuery<>(null, ProcessXPProperty.class);
 
     /*
      * Registry variables to persist
@@ -336,19 +344,57 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
         List<String> groupIDList = new ArrayList<>();
         int myPid = getProcessId();
 
-        // Get processes from WTS
+        // Structure we'll fill from native memory pointer
+        Pointer pProcessInfo = null;
+        WTS_PROCESS_INFO_EX[] processInfo = null;
+
+        // Get processes from WTS (post-XP)
         final PointerByReference ppProcessInfo = new PointerByReference();
         IntByReference pCount = new IntByReference(0);
-        if (!Wtsapi32.INSTANCE.WTSEnumerateProcessesEx(Wtsapi32.WTS_CURRENT_SERVER_HANDLE,
-                new IntByReference(Wtsapi32.WTS_PROCESS_INFO_LEVEL_1), Wtsapi32.WTS_ANY_SESSION, ppProcessInfo,
-                pCount)) {
-            LOG.error("Failed to enumerate Processes. Error code: {}", Kernel32.INSTANCE.GetLastError());
-            return new ArrayList<>(0);
+        try {
+            if (!Wtsapi32.INSTANCE.WTSEnumerateProcessesEx(Wtsapi32.WTS_CURRENT_SERVER_HANDLE,
+                    new IntByReference(Wtsapi32.WTS_PROCESS_INFO_LEVEL_1), Wtsapi32.WTS_ANY_SESSION, ppProcessInfo,
+                    pCount)) {
+                LOG.error("Failed to enumerate Processes. Error code: {}", Kernel32.INSTANCE.GetLastError());
+                return new ArrayList<>(0);
+            }
+            // extract the pointed-to pointer and create array
+            pProcessInfo = ppProcessInfo.getValue();
+            final WTS_PROCESS_INFO_EX processInfoRef = new WTS_PROCESS_INFO_EX(pProcessInfo);
+            processInfo = (WTS_PROCESS_INFO_EX[]) processInfoRef.toArray(pCount.getValue());
+        } catch (UnsatisfiedLinkError e) {
+            // On XP we can't use WTSEnumerateProcessesEx so we'll grab the same
+            // info from WMI and fake the array
+            StringBuilder sb = new StringBuilder(PROCESS_BASE_CLASS);
+            if (pids != null) {
+                boolean first = true;
+                for (Integer pid : pids) {
+                    if (first) {
+                        sb.append(" WHERE ProcessID=");
+                        first = false;
+                    } else {
+                        sb.append(" OR ProcessID=");
+                    }
+                    sb.append(pid);
+                }
+            }
+            PROCESS_QUERY_XP.setWmiClassName(sb.toString());
+            WmiResult<ProcessXPProperty> pseudoWTSResult = WmiQueryHandler.getInstance().queryWMI(PROCESS_QUERY_XP);
+            processInfo = new WTS_PROCESS_INFO_EX[pseudoWTSResult.getResultCount()];
+            for (int i = 0; i < pseudoWTSResult.getResultCount(); i++) {
+                processInfo[i].ProcessId = WmiUtil.getUint32(pseudoWTSResult, ProcessXPProperty.PROCESSID, i);
+                processInfo[i].pProcessName = WmiUtil.getString(pseudoWTSResult, ProcessXPProperty.NAME, i);
+                processInfo[i].KernelTime = new LARGE_INTEGER(
+                        WmiUtil.getUint64(pseudoWTSResult, ProcessXPProperty.KERNELMODETIME, i));
+                processInfo[i].UserTime = new LARGE_INTEGER(
+                        WmiUtil.getUint64(pseudoWTSResult, ProcessXPProperty.USERMODETIME, i));
+                processInfo[i].NumberOfThreads = WmiUtil.getUint32(pseudoWTSResult, ProcessXPProperty.THREADCOUNT, i);
+                // WMI Pagefile usage is in KB
+                processInfo[i].PagefileUsage = 1024
+                        * WmiUtil.getUint32(pseudoWTSResult, ProcessXPProperty.PAGEFILEUSAGE, i);
+                processInfo[i].HandleCount = WmiUtil.getUint32(pseudoWTSResult, ProcessXPProperty.HANDLECOUNT, i);
+            }
         }
-        // extract the pointed-to pointer and create array
-        final Pointer pProcessInfo = ppProcessInfo.getValue();
-        final WTS_PROCESS_INFO_EX processInfoRef = new WTS_PROCESS_INFO_EX(pProcessInfo);
-        final WTS_PROCESS_INFO_EX[] processInfo = (WTS_PROCESS_INFO_EX[]) processInfoRef.toArray(pCount.getValue());
 
         // Store a subset of processes in a list to later return.
         List<OSProcess> processList = new ArrayList<>();
@@ -446,7 +492,8 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
             processList.add(proc);
         }
         // Clean up memory allocated in C
-        if (!Wtsapi32.INSTANCE.WTSFreeMemoryEx(Wtsapi32.WTS_PROCESS_INFO_LEVEL_1, pProcessInfo, pCount.getValue())) {
+        if (pProcessInfo != null && !Wtsapi32.INSTANCE.WTSFreeMemoryEx(Wtsapi32.WTS_PROCESS_INFO_LEVEL_1, pProcessInfo,
+                pCount.getValue())) {
             LOG.error("Failed to Free Memory for Processes. Error code: {}", Kernel32.INSTANCE.GetLastError());
             return new ArrayList<>(0);
         }

--- a/oshi-core/src/main/java/oshi/software/os/windows/WindowsOperatingSystem.java
+++ b/oshi-core/src/main/java/oshi/software/os/windows/WindowsOperatingSystem.java
@@ -55,7 +55,6 @@ import com.sun.jna.platform.win32.WinError;
 import com.sun.jna.platform.win32.WinNT;
 import com.sun.jna.platform.win32.WinNT.HANDLE;
 import com.sun.jna.platform.win32.WinNT.HANDLEByReference;
-import com.sun.jna.platform.win32.WinNT.LARGE_INTEGER;
 import com.sun.jna.platform.win32.WinPerf.PERF_COUNTER_BLOCK;
 import com.sun.jna.platform.win32.WinPerf.PERF_COUNTER_DEFINITION;
 import com.sun.jna.platform.win32.WinPerf.PERF_DATA_BLOCK;
@@ -96,7 +95,7 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
 
     // Properties to get from WMI if WTSEnumerateProcesses doesn't work
     enum ProcessXPProperty {
-        PROCESSID, NAME, KERNELMODETIME, USERMODETIME, THREADCOUNT, PAGEFILEUSAGE, HANDLECOUNT;
+        PROCESSID, NAME, KERNELMODETIME, USERMODETIME, THREADCOUNT, PAGEFILEUSAGE, HANDLECOUNT, EXECUTABLEPATH;
     }
 
     private static final WmiQuery<ProcessXPProperty> PROCESS_QUERY_XP = new WmiQuery<>(null, ProcessXPProperty.class);
@@ -348,14 +347,17 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
         List<String> groupIDList = new ArrayList<>();
         int myPid = getProcessId();
 
-        // Structure we'll fill from native memory pointer
+        // Structure we'll fill from native memory pointer for Vista+
         Pointer pProcessInfo = null;
         WTS_PROCESS_INFO_EX[] processInfo = null;
+        IntByReference pCount = new IntByReference(0);
+
+        // WMI result we'll use for pre-Vista
+        WmiResult<ProcessXPProperty> processWmiResult = null;
 
         // Get processes from WTS (post-XP)
-        final PointerByReference ppProcessInfo = new PointerByReference();
-        IntByReference pCount = new IntByReference(0);
         if (IS_VISTA_OR_GREATER) {
+            final PointerByReference ppProcessInfo = new PointerByReference();
             if (!Wtsapi32.INSTANCE.WTSEnumerateProcessesEx(Wtsapi32.WTS_CURRENT_SERVER_HANDLE,
                     new IntByReference(Wtsapi32.WTS_PROCESS_INFO_LEVEL_1), Wtsapi32.WTS_ANY_SESSION, ppProcessInfo,
                     pCount)) {
@@ -383,28 +385,15 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
                 }
             }
             PROCESS_QUERY_XP.setWmiClassName(sb.toString());
-            WmiResult<ProcessXPProperty> pseudoWTSResult = WmiQueryHandler.getInstance().queryWMI(PROCESS_QUERY_XP);
-            processInfo = new WTS_PROCESS_INFO_EX[pseudoWTSResult.getResultCount()];
-            for (int i = 0; i < pseudoWTSResult.getResultCount(); i++) {
-                processInfo[i] = new WTS_PROCESS_INFO_EX();
-                processInfo[i].ProcessId = WmiUtil.getUint32(pseudoWTSResult, ProcessXPProperty.PROCESSID, i);
-                processInfo[i].pProcessName = WmiUtil.getString(pseudoWTSResult, ProcessXPProperty.NAME, i);
-                processInfo[i].KernelTime = new LARGE_INTEGER(
-                        WmiUtil.getUint64(pseudoWTSResult, ProcessXPProperty.KERNELMODETIME, i));
-                processInfo[i].UserTime = new LARGE_INTEGER(
-                        WmiUtil.getUint64(pseudoWTSResult, ProcessXPProperty.USERMODETIME, i));
-                processInfo[i].NumberOfThreads = WmiUtil.getUint32(pseudoWTSResult, ProcessXPProperty.THREADCOUNT, i);
-                // WMI Pagefile usage is in KB
-                processInfo[i].PagefileUsage = 1024
-                        * WmiUtil.getUint32(pseudoWTSResult, ProcessXPProperty.PAGEFILEUSAGE, i);
-                processInfo[i].HandleCount = WmiUtil.getUint32(pseudoWTSResult, ProcessXPProperty.HANDLECOUNT, i);
-            }
+            processWmiResult = WmiQueryHandler.getInstance().queryWMI(PROCESS_QUERY_XP);
         }
 
         // Store a subset of processes in a list to later return.
         List<OSProcess> processList = new ArrayList<>();
 
-        for (WTS_PROCESS_INFO_EX procInfo : processInfo) {
+        int procCount = IS_VISTA_OR_GREATER ? processInfo.length : processWmiResult.getResultCount();
+        for (int i = 0; i < procCount; i++) {
+
             // Skip if only updating a subset of pids, or if not in cache.
             // (Cache should have just been updated from registry so this will
             // only occur in a race condition for a just-started process.)
@@ -414,7 +403,8 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
             // ignore
             // the cache completely.
 
-            int pid = procInfo.ProcessId;
+            int pid = IS_VISTA_OR_GREATER ? processInfo[i].ProcessId
+                    : WmiUtil.getUint32(processWmiResult, ProcessXPProperty.PROCESSID, i);
             OSProcess proc = null;
             if (this.processMap.isEmpty()) {
                 if (pids != null && !pids.contains(pid)) {
@@ -422,7 +412,8 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
                 }
                 proc = new OSProcess();
                 proc.setProcessID(pid);
-                proc.setName(procInfo.pProcessName);
+                proc.setName(IS_VISTA_OR_GREATER ? processInfo[i].pProcessName
+                        : WmiUtil.getString(processWmiResult, ProcessXPProperty.NAME, i));
             } else {
                 proc = this.processMap.get(pid);
                 if (proc == null || pids != null && !pids.contains(pid)) {
@@ -436,8 +427,22 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
                 proc.setCurrentWorkingDirectory(cwd.isEmpty() ? "" : cwd.substring(0, cwd.length() - 1));
             }
 
-            proc.setKernelTime(procInfo.KernelTime.getValue() / 10000L);
-            proc.setUserTime(procInfo.UserTime.getValue() / 10000L);
+            if (IS_VISTA_OR_GREATER) {
+                WTS_PROCESS_INFO_EX procInfo = processInfo[i];
+                proc.setKernelTime(procInfo.KernelTime.getValue() / 10000L);
+                proc.setUserTime(procInfo.UserTime.getValue() / 10000L);
+                proc.setThreadCount(procInfo.NumberOfThreads);
+                proc.setVirtualSize(procInfo.PagefileUsage & 0xffff_ffffL);
+                proc.setOpenFiles(procInfo.HandleCount);
+            } else {
+                proc.setKernelTime(WmiUtil.getUint64(processWmiResult, ProcessXPProperty.KERNELMODETIME, i) / 10000L);
+                proc.setUserTime(WmiUtil.getUint64(processWmiResult, ProcessXPProperty.USERMODETIME, i) / 10000L);
+                proc.setThreadCount(WmiUtil.getUint32(processWmiResult, ProcessXPProperty.THREADCOUNT, i));
+                // WMI Pagefile usage is in KB
+                proc.setVirtualSize(1024
+                        * (WmiUtil.getUint32(processWmiResult, ProcessXPProperty.PAGEFILEUSAGE, i) & 0xffff_ffffL));
+                proc.setOpenFiles(WmiUtil.getUint32(processWmiResult, ProcessXPProperty.HANDLECOUNT, i));
+            }
 
             // Get a handle to the process for various extended info. Only gets
             // current user unless running as administrator
@@ -446,8 +451,9 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
             if (pHandle != null) {
                 // Full path
                 final HANDLEByReference phToken = new HANDLEByReference();
-                try {
-                    proc.setPath(Kernel32Util.QueryFullProcessImageName(pHandle, 0));
+                try {// EXECUTABLEPATH
+                    proc.setPath(IS_VISTA_OR_GREATER ? Kernel32Util.QueryFullProcessImageName(pHandle, 0)
+                            : WmiUtil.getString(processWmiResult, ProcessXPProperty.EXECUTABLEPATH, i));
                     if (Advapi32.INSTANCE.OpenProcessToken(pHandle, WinNT.TOKEN_DUPLICATE | WinNT.TOKEN_QUERY,
                             phToken)) {
                         Account account = Advapi32Util.getTokenAccount(phToken.getValue());
@@ -490,15 +496,14 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
             // State and possibly roll up.
             proc.setState(OSProcess.State.RUNNING);
 
-            proc.setThreadCount(procInfo.NumberOfThreads);
-            proc.setVirtualSize(procInfo.PagefileUsage & 0xffff_ffffL);
-
-            proc.setOpenFiles(procInfo.HandleCount);
             processList.add(proc);
         }
-        // Clean up memory allocated in C
+        // Clean up memory allocated in C (only Vista+ but null pointer
+        // effectively tests)
         if (pProcessInfo != null && !Wtsapi32.INSTANCE.WTSFreeMemoryEx(Wtsapi32.WTS_PROCESS_INFO_LEVEL_1, pProcessInfo,
-                pCount.getValue())) {
+                pCount.getValue()))
+
+        {
             LOG.error("Failed to Free Memory for Processes. Error code: {}", Kernel32.INSTANCE.GetLastError());
             return new ArrayList<>(0);
         }

--- a/oshi-core/src/main/java/oshi/software/os/windows/WindowsOperatingSystem.java
+++ b/oshi-core/src/main/java/oshi/software/os/windows/WindowsOperatingSystem.java
@@ -386,6 +386,7 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
             WmiResult<ProcessXPProperty> pseudoWTSResult = WmiQueryHandler.getInstance().queryWMI(PROCESS_QUERY_XP);
             processInfo = new WTS_PROCESS_INFO_EX[pseudoWTSResult.getResultCount()];
             for (int i = 0; i < pseudoWTSResult.getResultCount(); i++) {
+                processInfo[i] = new WTS_PROCESS_INFO_EX();
                 processInfo[i].ProcessId = WmiUtil.getUint32(pseudoWTSResult, ProcessXPProperty.PROCESSID, i);
                 processInfo[i].pProcessName = WmiUtil.getString(pseudoWTSResult, ProcessXPProperty.NAME, i);
                 processInfo[i].KernelTime = new LARGE_INTEGER(

--- a/oshi-core/src/main/java/oshi/software/os/windows/WindowsOperatingSystem.java
+++ b/oshi-core/src/main/java/oshi/software/os/windows/WindowsOperatingSystem.java
@@ -69,6 +69,7 @@ import com.sun.jna.platform.win32.COM.WbemcliUtil.WmiResult;
 import com.sun.jna.ptr.IntByReference;
 import com.sun.jna.ptr.PointerByReference;
 
+import oshi.jna.platform.windows.VersionHelpers;
 import oshi.software.common.AbstractOperatingSystem;
 import oshi.software.os.FileSystem;
 import oshi.software.os.NetworkParams;
@@ -99,6 +100,9 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
     }
 
     private static final WmiQuery<ProcessXPProperty> PROCESS_QUERY_XP = new WmiQuery<>(null, ProcessXPProperty.class);
+
+    // Is AddEnglishCounter available?
+    private static final boolean IS_VISTA_OR_GREATER = VersionHelpers.IsWindowsVistaOrGreater();
 
     /*
      * Registry variables to persist
@@ -351,7 +355,7 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
         // Get processes from WTS (post-XP)
         final PointerByReference ppProcessInfo = new PointerByReference();
         IntByReference pCount = new IntByReference(0);
-        try {
+        if (IS_VISTA_OR_GREATER) {
             if (!Wtsapi32.INSTANCE.WTSEnumerateProcessesEx(Wtsapi32.WTS_CURRENT_SERVER_HANDLE,
                     new IntByReference(Wtsapi32.WTS_PROCESS_INFO_LEVEL_1), Wtsapi32.WTS_ANY_SESSION, ppProcessInfo,
                     pCount)) {
@@ -362,9 +366,9 @@ public class WindowsOperatingSystem extends AbstractOperatingSystem {
             pProcessInfo = ppProcessInfo.getValue();
             final WTS_PROCESS_INFO_EX processInfoRef = new WTS_PROCESS_INFO_EX(pProcessInfo);
             processInfo = (WTS_PROCESS_INFO_EX[]) processInfoRef.toArray(pCount.getValue());
-        } catch (UnsatisfiedLinkError e) {
-            // On XP we can't use WTSEnumerateProcessesEx so we'll grab the same
-            // info from WMI and fake the array
+        } else {
+            // Pre-Vista we can't use WTSEnumerateProcessesEx so we'll grab the
+            // same info from WMI and fake the array
             StringBuilder sb = new StringBuilder(PROCESS_BASE_CLASS);
             if (pids != null) {
                 boolean first = true;

--- a/oshi-core/src/main/java/oshi/util/platform/windows/PdhUtilXP.java
+++ b/oshi-core/src/main/java/oshi/util/platform/windows/PdhUtilXP.java
@@ -1,0 +1,102 @@
+/**
+ * OSHI (https://github.com/oshi/oshi)
+ *
+ * Copyright (c) 2010 - 2019 The OSHI Project Team:
+ * https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.util.platform.windows;
+
+import com.sun.jna.Memory; //NOSONAR
+import com.sun.jna.Native;
+import com.sun.jna.platform.win32.Pdh;
+import com.sun.jna.platform.win32.PdhUtil.PdhException;
+import com.sun.jna.platform.win32.WinDef.DWORD;
+import com.sun.jna.platform.win32.WinDef.DWORDByReference;
+import com.sun.jna.platform.win32.WinError;
+
+/**
+ * TODO: This class is compatible with Windows XP and will be removed when the
+ * JNA version of it is released.
+ */
+public class PdhUtilXP {
+    private static final int CHAR_TO_BYTES = Boolean.getBoolean("w32.ascii") ? 1 : Native.WCHAR_SIZE;
+
+    // Windows XP requires non-null buffer size for PdhLookupPerfNameByIndex.
+    // Use this as a flag to determine whether we need to pass a buffer to the
+    // function. -1 is untested, 0 means null buffer is allowed, and 1 means a
+    // buffer is required.
+    private static int minPerfNameBufferSize = -1;
+
+    private static final int PDH_INSUFFICIENT_BUFFER = 0xC0000BC2;
+
+    /**
+     * Utility method to call Pdh's PdhLookupPerfNameByIndex that allocates the
+     * required memory for the szNameBuffer parameter based on the type mapping
+     * used, calls to PdhLookupPerfNameByIndex, and returns the received string.
+     * 
+     * @param szMachineName
+     *            Null-terminated string that specifies the name of the computer
+     *            where the specified performance object or counter is located.
+     *            The computer name can be specified by the DNS name or the IP
+     *            address. If NULL, the function uses the local computer.
+     * @param dwNameIndex
+     *            Index of the performance object or counter.
+     * @return Returns the name of the performance object or counter.
+     */
+    public static String PdhLookupPerfNameByIndex(String szMachineName, int dwNameIndex) {
+        // If we haven't established whether buffer is required, test for that
+        if (minPerfNameBufferSize < 0) {
+            int result = Pdh.INSTANCE.PdhLookupPerfNameByIndex(szMachineName, dwNameIndex, null,
+                    new DWORDByReference(new DWORD(0)));
+            if (result == WinError.ERROR_SUCCESS || result == Pdh.PDH_MORE_DATA) {
+                minPerfNameBufferSize = 0;
+            } else {
+                minPerfNameBufferSize = 1;
+            }
+        }
+
+        // Call once to get required buffer size
+        DWORDByReference pcchNameBufferSize = new DWORDByReference(new DWORD(minPerfNameBufferSize));
+        Memory mem = minPerfNameBufferSize > 0 ? new Memory(minPerfNameBufferSize) : null;
+        int result = Pdh.INSTANCE.PdhLookupPerfNameByIndex(szMachineName, dwNameIndex, mem, pcchNameBufferSize);
+        if (result != WinError.ERROR_SUCCESS && result != Pdh.PDH_MORE_DATA && result != PDH_INSUFFICIENT_BUFFER) {
+            throw new PdhException(result);
+        }
+
+        // Can't allocate 0 memory
+        if (pcchNameBufferSize.getValue().intValue() < 1) {
+            return "";
+        }
+        // Allocate buffer and call again
+        mem = new Memory(pcchNameBufferSize.getValue().intValue() * CHAR_TO_BYTES);
+        result = Pdh.INSTANCE.PdhLookupPerfNameByIndex(szMachineName, dwNameIndex, mem, pcchNameBufferSize);
+
+        if (result != WinError.ERROR_SUCCESS) {
+            throw new PdhException(result);
+        }
+
+        // Convert buffer to Java String
+        if (CHAR_TO_BYTES == 1) {
+            return mem.getString(0);
+        } else {
+            return mem.getWideString(0);
+        }
+    }
+}

--- a/oshi-core/src/main/java/oshi/util/platform/windows/PdhUtilXP.java
+++ b/oshi-core/src/main/java/oshi/util/platform/windows/PdhUtilXP.java
@@ -31,6 +31,8 @@ import com.sun.jna.platform.win32.WinDef.DWORD;
 import com.sun.jna.platform.win32.WinDef.DWORDByReference;
 import com.sun.jna.platform.win32.WinError;
 
+import oshi.jna.platform.windows.VersionHelpers;
+
 /**
  * TODO: This class is compatible with Windows XP and will be removed when the
  * JNA version of it is released.
@@ -38,11 +40,8 @@ import com.sun.jna.platform.win32.WinError;
 public class PdhUtilXP {
     private static final int CHAR_TO_BYTES = Boolean.getBoolean("w32.ascii") ? 1 : Native.WCHAR_SIZE;
 
-    // Windows XP requires non-null buffer size for PdhLookupPerfNameByIndex.
-    // Use this as a flag to determine whether we need to pass a buffer to the
-    // function. -1 is untested, 0 means null buffer is allowed, and 1 means a
-    // buffer is required.
-    private static int minPerfNameBufferSize = -1;
+    // Can I pass null buffer to PdhLookupPerfNameByIndex?
+    private static final boolean IS_VISTA_OR_GREATER = VersionHelpers.IsWindowsVistaOrGreater();
 
     private static final int PDH_INSUFFICIENT_BUFFER = 0xC0000BC2;
 
@@ -61,20 +60,9 @@ public class PdhUtilXP {
      * @return Returns the name of the performance object or counter.
      */
     public static String PdhLookupPerfNameByIndex(String szMachineName, int dwNameIndex) {
-        // If we haven't established whether buffer is required, test for that
-        if (minPerfNameBufferSize < 0) {
-            int result = Pdh.INSTANCE.PdhLookupPerfNameByIndex(szMachineName, dwNameIndex, null,
-                    new DWORDByReference(new DWORD(0)));
-            if (result == WinError.ERROR_SUCCESS || result == Pdh.PDH_MORE_DATA) {
-                minPerfNameBufferSize = 0;
-            } else {
-                minPerfNameBufferSize = 1;
-            }
-        }
-
         // Call once to get required buffer size
-        DWORDByReference pcchNameBufferSize = new DWORDByReference(new DWORD(minPerfNameBufferSize));
-        Memory mem = minPerfNameBufferSize > 0 ? new Memory(minPerfNameBufferSize) : null;
+        DWORDByReference pcchNameBufferSize = new DWORDByReference(new DWORD(IS_VISTA_OR_GREATER ? 1 : 0));
+        Memory mem = IS_VISTA_OR_GREATER ? null : new Memory(1);
         int result = Pdh.INSTANCE.PdhLookupPerfNameByIndex(szMachineName, dwNameIndex, mem, pcchNameBufferSize);
         if (result != WinError.ERROR_SUCCESS && result != Pdh.PDH_MORE_DATA && result != PDH_INSUFFICIENT_BUFFER) {
             throw new PdhException(result);

--- a/oshi-core/src/main/java/oshi/util/platform/windows/PerfDataUtil.java
+++ b/oshi-core/src/main/java/oshi/util/platform/windows/PerfDataUtil.java
@@ -45,6 +45,7 @@ import com.sun.jna.platform.win32.WinError;
 import com.sun.jna.platform.win32.WinNT;
 import com.sun.jna.platform.win32.WinNT.HANDLEByReference;
 
+import oshi.jna.platform.windows.VersionHelpers;
 import oshi.util.Util;
 
 /**
@@ -79,6 +80,9 @@ public class PerfDataUtil {
     private static final Map<PerfCounter, HANDLEByReference> counterMap = new HashMap<>();
     private static final Map<String, HANDLEByReference> queryMap = new HashMap<>();
     private static final Set<String> disabledQueries = new HashSet<>();
+
+    // Is AddEnglishCounter available?
+    private static final boolean IS_VISTA_OR_GREATER = VersionHelpers.IsWindowsVistaOrGreater();
 
     public class PerfCounter {
         private String object;
@@ -329,11 +333,11 @@ public class PerfDataUtil {
      */
     private static boolean addCounter(WinNT.HANDLEByReference query, String path, WinNT.HANDLEByReference p) {
         int pdhAddCounterError;
-        try {
+        if (IS_VISTA_OR_GREATER) {
             // Localized version for Vista+
             pdhAddCounterError = PDH.PdhAddEnglishCounter(query.getValue(), path, PZERO, p);
-        } catch (UnsatisfiedLinkError e) {
-            // Will occur on Windows XP, only will work for English counters
+        } else {
+            // Only will work for English counters
             pdhAddCounterError = PDH.PdhAddCounter(query.getValue(), path, PZERO, p);
         }
         if (pdhAddCounterError != WinError.ERROR_SUCCESS && LOG.isWarnEnabled()) {

--- a/oshi-core/src/main/java/oshi/util/platform/windows/PerfDataUtil.java
+++ b/oshi-core/src/main/java/oshi/util/platform/windows/PerfDataUtil.java
@@ -377,13 +377,22 @@ public class PerfDataUtil {
      */
     private static long updateQueryTimestamp(WinNT.HANDLEByReference query) {
         LONGLONGByReference pllTimeStamp = new LONGLONGByReference();
-        int ret = PDH.PdhCollectQueryDataWithTime(query.getValue(), pllTimeStamp);
+        int ret;
+        if (IS_VISTA_OR_GREATER) {
+            ret = PDH.PdhCollectQueryDataWithTime(query.getValue(), pllTimeStamp);
+        } else {
+            ret = PDH.PdhCollectQueryData(query.getValue());
+        }
         // Due to race condition, initial update may fail with PDH_NO_DATA.
         int retries = 0;
         while (ret == PdhMsg.PDH_NO_DATA && retries++ < 3) {
             // Exponential fallback.
             Util.sleep(1 << retries);
-            ret = PDH.PdhCollectQueryDataWithTime(query.getValue(), pllTimeStamp);
+            if (IS_VISTA_OR_GREATER) {
+                ret = PDH.PdhCollectQueryDataWithTime(query.getValue(), pllTimeStamp);
+            } else {
+                ret = PDH.PdhCollectQueryData(query.getValue());
+            }
         }
         if (ret != WinError.ERROR_SUCCESS) {
             if (LOG.isWarnEnabled()) {
@@ -392,7 +401,13 @@ public class PerfDataUtil {
             return 0;
         }
         // Perf Counter timestamp is in local time
-        return filetimeToUtcMs(pllTimeStamp.getValue().longValue(), true);
+        if (IS_VISTA_OR_GREATER) {
+            return filetimeToUtcMs(pllTimeStamp.getValue().longValue(), true);
+        } else {
+            // Approximate timestamp
+            return System.currentTimeMillis();
+        }
+
     }
 
     /**

--- a/oshi-core/src/main/java/oshi/util/platform/windows/PerfDataUtil.java
+++ b/oshi-core/src/main/java/oshi/util/platform/windows/PerfDataUtil.java
@@ -328,7 +328,14 @@ public class PerfDataUtil {
      * @return
      */
     private static boolean addCounter(WinNT.HANDLEByReference query, String path, WinNT.HANDLEByReference p) {
-        int pdhAddCounterError = PDH.PdhAddEnglishCounter(query.getValue(), path, PZERO, p);
+        int pdhAddCounterError;
+        try {
+            // Localized version for Vista+
+            pdhAddCounterError = PDH.PdhAddEnglishCounter(query.getValue(), path, PZERO, p);
+        } catch (UnsatisfiedLinkError e) {
+            // Will occur on Windows XP, only will work for English counters
+            pdhAddCounterError = PDH.PdhAddCounter(query.getValue(), path, PZERO, p);
+        }
         if (pdhAddCounterError != WinError.ERROR_SUCCESS && LOG.isWarnEnabled()) {
             LOG.warn("Failed to add PDH Counter: {}, Error code: {}", path,
                     String.format(HEX_ERROR_FMT, pdhAddCounterError));

--- a/oshi-core/src/main/java/oshi/util/platform/windows/WmiQueryHandler.java
+++ b/oshi-core/src/main/java/oshi/util/platform/windows/WmiQueryHandler.java
@@ -1,20 +1,25 @@
 /**
- * Oshi (https://github.com/oshi/oshi)
+ * OSHI (https://github.com/oshi/oshi)
  *
- * Copyright (c) 2010 - 2018 The Oshi Project Team
- *
- * All rights reserved. This program and the accompanying materials
- * are made available under the terms of the Eclipse Public License v1.0
- * which accompanies this distribution, and is available at
- * http://www.eclipse.org/legal/epl-v10.html
- *
- * Maintainers:
- * dblock[at]dblock[dot]org
- * widdis[at]gmail[dot]com
- * enrico.bianchi[at]gmail[dot]com
- *
- * Contributors:
+ * Copyright (c) 2010 - 2019 The OSHI Project Team:
  * https://github.com/oshi/oshi/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
  */
 package oshi.util.platform.windows;
 

--- a/oshi-core/src/test/java/oshi/software/os/OperatingSystemTest.java
+++ b/oshi-core/src/test/java/oshi/software/os/OperatingSystemTest.java
@@ -77,7 +77,7 @@ public class OperatingSystemTest {
         assertTrue(proc.getParentProcessID() > 0);
         assertTrue(proc.getThreadCount() > 0);
         assertTrue(proc.getPriority() >= -20 && proc.getPriority() <= 128);
-        assertTrue(proc.getVirtualSize() >= proc.getResidentSetSize());
+        assertTrue(proc.getVirtualSize() >= 0);
         assertTrue(proc.getResidentSetSize() >= 0);
         assertTrue(proc.getKernelTime() >= 0);
         assertTrue(proc.getUserTime() >= 0);

--- a/oshi-core/src/test/java/oshi/util/UtilTest.java
+++ b/oshi-core/src/test/java/oshi/util/UtilTest.java
@@ -38,12 +38,12 @@ public class UtilTest {
     public void testSleep() {
         long now = System.currentTimeMillis();
         Util.sleep(100);
-        assertTrue(System.currentTimeMillis() - now >= 100);
+        assertTrue(System.currentTimeMillis() - now >= 99);
 
         now = System.currentTimeMillis();
         long then = now + 100;
         Util.sleepAfter(then, 100);
-        assertTrue(System.currentTimeMillis() - now >= 200);
+        assertTrue(System.currentTimeMillis() - now >= 199);
 
         now = System.currentTimeMillis();
         then = now - 550;

--- a/oshi-json/src/test/java/oshi/json/software/os/OperatingSystemTest.java
+++ b/oshi-json/src/test/java/oshi/json/software/os/OperatingSystemTest.java
@@ -77,7 +77,7 @@ public class OperatingSystemTest {
         assertTrue(proc.getParentProcessID() > 0);
         assertTrue(proc.getThreadCount() > 0);
         assertTrue(proc.getPriority() >= -20 && proc.getPriority() <= 128);
-        assertTrue(proc.getVirtualSize() >= proc.getResidentSetSize());
+        assertTrue(proc.getVirtualSize() >= 0);
         assertTrue(proc.getResidentSetSize() >= 0);
         assertTrue(proc.getKernelTime() >= 0);
         assertTrue(proc.getUserTime() >= 0);

